Sources/NIOIMAPCore/Grammar/Mailbox/MailboxName.swift

```diff
@@ -197,9 +197,12 @@ public struct MailboxName {
     /// `true` if the internal storage reads "INBOX"
     /// otherwise `false`
     public var isInbox: Bool {
-        bytes.lazy.map { $0 & 0xDF }.elementsEqual("INBOX".utf8)
+        hashValue == MailboxName.inboxHashValue &&
+            bytes.lazy.map { $0 & 0xDF }.elementsEqual("INBOX".utf8)
```

Comment:
Do we really need the lazy check anymore if we have the precomputed hash?

Reply:
Probably not, but it doesn’t hurt. Otherwise we’d do `$0 & 0xdf` on everything…? We could do
```
hashValue == MailboxName.inboxHashValue &&
    bytes.count == 5 &&
    bytes.map { $0 & 0xDF }.elementsEqual("INBOX".utf8)
```
?

Reply:
Your call :) I was just wondering about the extra (although tiny) amount of extra code that maybe we don't need... Hash collisions should be very rare, so I thought maybe using only the `inboxHashValue` would do the trick. Whatever you recommend though!